File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int ptrOnS=0;
        int ptrOnT=0;
        int count = s.length();

        while(ptrOnS<s.length() && ptrOnT<t.length())
        {
            if(s[ptrOnS] == t[ptrOnT])
            {
                count--;
                ptrOnS++;
            }
            ptrOnT++;
        }
        return (count==0);
    }
};

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        char[] sChar = s.toCharArray();
        char[] tChar = t.toCharArray();

        int sStart = 0;
        int tStart = 0;

        while ((sStart < sChar.length) && (tStart < tChar.length)) {
            if (sChar[sStart] == tChar[tStart]) {
                sStart++;
            }
            tStart++;
        }
        if (sStart == sChar.length) {
            return true;
        }
        return false;
    }
}